src/render.js

```diff
@@ -1,30 +1,38 @@
 const error = require('./error.js');
 const Exp = require('./expression.js');
 const assert = require('assert');
+const Promise = require('any-promise');
 
 var render = {
 
     renderTemplates: function(templates, scope) {
         assert(scope, 'unable to evalTemplates: scope undefined');
-        var html = '',
-            partial;
-        templates.some(template => {
+        var htmlBlocks = [],
+            partial,
+            promises = [];
+        templates.some((template, index) => {
```

Comment:
Since `render` returns Promises now, all tags require `renderTemplates` will need this tech to re-construct :)